src/WebJobs.Script.WebHost/Controllers/FunctionsController.cs

```diff
@@ -100,4 +100,33 @@ public FunctionsController(WebScriptHostManager scriptHostManager, WebHookReceiv
             return response;
         }
     }
+    
+    // Custom web hooks for extensions. 
+    public class HookController : ApiController
+    {
+        private readonly WebScriptHostManager _scriptHostManager;
+
+        public HookController(WebScriptHostManager scriptHostManager)
+        {
+            _scriptHostManager = scriptHostManager;
+        }
+                
+        [Route("hook/{name}")]
+        [HttpGet]
+        [HttpPost]
+        [AllowAnonymous]
+        public async Task<HttpResponseMessage> ExtensionHook(string name, CancellationToken token)
+        {
+            var host = this._scriptHostManager.Instance;
+
+            IAsyncConverter<HttpRequestMessage, HttpResponseMessage> hook;
+            if (host.CustomHttpHandlers.TryGetValue(name, out hook))
```

Comment:
See other comments below about moving this http stuff into WebScriptHostManager where it belongs. Then instead of this code, you'd call someting like `_scriptHostManager.HandleExtensionWebHookAsync`, keeping all these internals encapsulated there.